python-avd/pyavd/_eos_designs/structured_config/network_services/router_internet_exit.py

```diff
@@ -19,28 +21,200 @@ class RouterInternetExitMixin(Protocol):
+        """Object to keep track of tunnel metadata for Zscaler Intermet Exit Policy."""
+
+        for wan_interface in local_wan_l3_interfaces:
+            if not wan_interface.peer_ip:
+                msg = f"The configured internet-exit policy requires `peer_ip` configured under the WAN Interface {wan_interface.name}."
+                raise AristaAvdInvalidInputsError(msg)
+
+            interface_policy_config = wan_interface.cv_pathfinder_internet_exit.policies[internet_exit_policy.name]
+
+            if interface_policy_config.tunnel_interface_numbers is None:
+                msg = (
+                    f"{wan_interface.name}.cv_pathfinder_internet_exit.policies[{internet_exit_policy.name}]."
+                    "tunnel_interface_numbers needs to be set, when using wan interface for zscaler type internet exit."
+                )
+                raise AristaAvdInvalidInputsError(msg)
+
+            tunnel_id_range: list[int] = range_expand(interface_policy_config.tunnel_interface_numbers)
+
+            zscaler_endpoints = (self._zscaler_endpoints.primary, self._zscaler_endpoints.secondary, self._zscaler_endpoints.tertiary)
+            for index, zscaler_endpoint in enumerate(zscaler_endpoints):
+                if not zscaler_endpoint:
+                    continue
+
+                tunnel_id = tunnel_id_range[index]
+                preference = ("primary", "secondary", "tertiary")[index]
+                # PRI, SEC, TER used for groups
+                # TODO: consider if we should use DC names as group suffix.
+                suffix = preference[0:3].upper()
+
+                connection_name = f"IE-Tunnel{tunnel_id}"
+                description = f"Internet Exit {internet_exit_policy.name} {suffix}"
+
+                self.set_zscaler_ie_tunnel_interface(
+                    tunnel_id=tunnel_id,
+                    description=description,
+                    source_interface=wan_interface.name,
+                    destination=zscaler_endpoint.ip_address,
+                    ipsec_profile=f"IE-{internet_exit_policy.name}-PROFILE",
+                )
+                self.set_zscaler_ie_monitor_connectivity(
+                    tunnel_id=tunnel_id, monitor_name=f"IE-Tunnel{tunnel_id}", description=description, monitor_host=zscaler_endpoint.ip_address
+                )
+                self.set_zscaler_ie_connection_static_route(
+                    destination_ip=zscaler_endpoint.ip_address, name=f"IE-ZSCALER-{suffix}", next_hop=wan_interface.peer_ip
+                )
+                self.set_zscaler_ie_router_service_insertion(monitor_name=connection_name, tunnel_id=tunnel_id)
+
+                # set metadata
+                metadata_tunnels.append_new(
+                    name=f"Tunnel{tunnel_id}",
+                    preference="Preferred" if preference == "primary" else "Alternate",
+                    # TODO: type error
+                    endpoint=zscaler_endpoint,
```

Comment:
type issue. We should proably create a new object or (after checking they are perfectly compatible) use _cast_as.

Reply:
they are perfectly compatible, using _cast_as